src/components/component-playground.jsx

```diff
@@ -525,5 +548,40 @@ module.exports = React.createClass({
       orientation: contentNode.offsetHeight > contentNode.offsetWidth ?
                    'portrait' : 'landscape'
     });
+  },
+
+  _getFilteredComponents() {
+    return _.mapValues(this.props.components, function(fixtures) {
+      return _.assign({}, fixtures, {
+      // Only select fixtures matching search text and rebuild the object.
+        fixtures: _.pick(fixtures.fixtures,
+            function(fixtureProps, fixtureName) {
+              return fixtureName.indexOf(this.state.searchText) !== -1;
+            }.bind(this)
+        )
+      });
+    }.bind(this));
+  },
+
+  _getFilteredFixtures() {
+    // Get components which match search text by one or more fixture names.
+    var matchedByFixtureName = _.pick(this._getFilteredComponents(),
+        // Ignore the component if no fixture matched.
+        function(fixtures) {
+          return !_.isEmpty(fixtures.fixtures);
+        }
+    );
+
+    // Get components which match search text by component name and the
+    // currently selected one.
+    var matchedByComponentName = _.pick(this.props.components,
+        function(fixtureProps, componentName) {
+          return componentName.indexOf(this.state.searchText) !== -1 ||
+                 componentName === this.props.component;
```

Comment:
Doing this will make the selected component always show up first.
Then, selecting a new component will make it 'snap to the top':

![snap-first](https://cloud.githubusercontent.com/assets/20299983/17554073/d51c84d6-5f11-11e6-8153-630f5f89dfe1.gif)

Is this really what we want?